tests/tpu/test_compilation.py

```diff
@@ -0,0 +1,18 @@
+import glob
+import os
+import runpy
+import tempfile
+
+import depyf
+
+temp_dir = tempfile.mkdtemp()
+with depyf.prepare_debug(temp_dir):
+    cur_dir = os.path.dirname(__file__)
+    parent_dir = os.path.dirname(cur_dir)
+    root_dir = os.path.dirname(parent_dir)
+    example_file = os.path.join(root_dir, "examples",
+                                "offline_inference_tpu.py")
+    runpy.run_path(example_file)
+
+compiled_code = glob.glob(os.path.join(temp_dir, "__transformed_code*.py"))
+assert len(compiled_code) == 2
```

Comment:
nit: worth leaving a comment on why 2 is expected if this begins to fail in future work

Reply:
good idea, added in [db73048](https://github.com/vllm-project/vllm/pull/7796/commits/db73048e1f4c555608043ffa72cf6699e9e8d28b)